### PR TITLE
Use parameters from storage instead of ensemble_config

### LINKED ISSUE #5566: Load parameter configuration from storage
When we are loading data from runpath we are using the user configuration:

https://github.com/equinor/ert/blob/66a8030f0d5404b4b6b725e82dbcc361955516e1/src/ert/callbacks.py#LL68C39-L68C39

however we are also internalizing the same configuration in storage. We should use the parameter configuration from storage instead of directly from the user configuration to avoid the two comming out of sync.

We have a `to_dict` function that we use to write the objects to disk, we should also have a `from_dict` so we can restore them from disk.

https://github.com/equinor/ert/blob/66a8030f0d5404b4b6b725e82dbcc361955516e1/src/ert/_c_wrappers/enkf/config/parameter_config.py#L39

### PR DESCRIPTION
**Pull request by oyvindeide**
**Issue**
Resolves #5566 

**Approach**
_Short description of the approach_

## Pre review checklist

- [x] Added appropriate release note label
- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [x] Updated documentation
- [x] Ensured new behaviour is tested

Adding labels helps the maintainers when writing release notes. This is the [list of release note labels](https://github.com/equinor/ert/labels?q=release-notes).

### CODE REVIEW

File: src/ert/_c_wrappers/enkf/config/field_config.py [hunk 1]
```diff
@@ -54,8 +52,8 @@ def read_from_runpath(
         trans = self.input_transformation
         data_transformed = field_transform(data, trans)
         ds = xr.Dataset({"values": (["x", "y", "z"], data_transformed)})
-        ensemble.save_parameters(key, real_nr, ds)
         _logger.debug(f"load() time_used {(time.perf_counter() - t):.4f}s")
+        return ds
```
Comment (frode-aarstad) [inline src/ert/_c_wrappers/enkf/config/field_config.py:56]: Maybe add return type to funcion

File: src/ert/_c_wrappers/enkf/config/gen_kw_config.py [hunk 2]
```diff
@@ -57,23 +54,21 @@ def sample_or_load(
...
         )
 
-        dataset = xr.Dataset(
+        return xr.Dataset(
             {
                 "values": ("names", parameter_value),
                 "transformed_values": ("names", self.transform(parameter_value)),
                 "names": keys,
             }
         )
```
Comment (frode-aarstad) [inline src/ert/_c_wrappers/enkf/config/gen_kw_config.py:66]: Add return type to function

**Review by frode-aarstad — COMMENTED** [review]
Looks good, only lacking some return types for functions

File: tests/unit_tests/scenarios/test_summary_response.py [hunk 3]
```diff
@@ -67,7 +62,10 @@ def setup_configuration(tmpdir):
         run_sim(ref_time, 0.5)
         ert_config = ErtConfig.from_file("config.ert")
         ert = EnKFMain(ert_config)
-        yield ert
+        ensemble = storage.create_experiment(
+            parameters=ert_config.ensemble_config.parameter_configuration
+        ).create_ensemble(ensemble_size=100, name="prior")
+        yield ert, ensemble
```
Comment (pinkwah) [inline tests/unit_tests/scenarios/test_summary_response.py:68]: Stylistic nitpick, but you can keep `prior_ensemble` as a separate fixture by having it depend on `setup_configuration`. This fixture will be run exactly once per test, so the tests that depend on both `setup_configuration` and `prior_ensemble` will get the same exact ERT object as `prior_ensemble` did.

**Review by pinkwah — APPROVED** [review]
Some minor stuff but nothing too important.

I also agree with @frode-aarstad 's comments, but they fall into the "minor stuff" category. Nice to have but we can fix it later.

**Pull request merged**